test/ruby/test_econv.rb

```diff
@@ -22,8 +22,8 @@ def assert_econv(converted, eres, obuf_bytesize, ec, consumed, rest, opts=nil)
 
   def assert_errinfo(e_res, e_enc1, e_enc2, e_error_bytes, e_readagain_bytes, ec)
     assert_equal([e_res, e_enc1, e_enc2,
-                  e_error_bytes && e_error_bytes.dup.force_encoding("ASCII-8BIT"),
-                  e_readagain_bytes && e_readagain_bytes.dup.force_encoding("ASCII-8BIT")],
+                  e_error_bytes&.dup&.force_encoding("ASCII-8BIT"),
+                  e_readagain_bytes&.dup&.force_encoding("ASCII-8BIT")],
```

Comment:
That will raise an error in a `nil` case:
```ruby
e_error_bytes = nil
# => nil
e_error_bytes&.dup&.force_encoding
# => nil
e_error_bytes&.dup.force_encoding
# => NoMethodError: undefined method `force_encoding' for nil:NilClass
e_error_bytes && e_error_bytes.dup.force_encoding
# => nil
```
I can revert the change entirely if you want